cmd/promtool/resource.go

```diff
@@ -41,7 +41,7 @@ func newResourceSaver(cfg resourceSaverConfig) *resourceSaver {
 	if err != nil {
 		panic(err)
 	}
-	tw, err := newArchiveWriter(archiveWriterConfig{archiveName: cfg.tarballName})
+	a, err := newArchiver(archiverConfig{archiveName: cfg.tarballName})
```

Comment:
I personally prefer more descriptive names - `archiver`, this would also make more  sense when you initialise the struct below , but not a deal breaker.

Reply:
Agree, since we removed the field name below.